src/include/distributed/remote_transaction.h

```diff
@@ -50,7 +50,9 @@ typedef enum
 	/* transaction commit */
 	REMOTE_TRANS_1PC_COMMITTING,
 	REMOTE_TRANS_2PC_COMMITTING,
-	REMOTE_TRANS_COMMITTED
+	REMOTE_TRANS_COMMITTED,
+
+	REMOTE_TRANS_OUTSIDE_COORDINATED_TX,
```

Comment:
This state is meant to reflect remote session state (e.g. we keep track of the fact that we sent BEGIN), so not sure this is the right approach, since this state says more about how we intend to use the connection and is lost as soon as we BEGIN.

I think it is ok for these connections to be in InProgressTransactions , and conceptually it makes sense, but we should remove them from InProgressTransactions in FinishRemoteTransactionCommit/Abort.